src/mpid/ch4/shm/ipc/gpu/gpu_init.c

```diff
@@ -43,10 +43,10 @@ static void ipc_handle_free_hook(void *dptr)
     MPIR_FUNC_ENTER;
 
     if (MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE) {
-        mpl_err = MPL_gpu_get_buffer_bounds(dptr, &pbase, &len);
+        MPIR_GPU_query_pointer_attr(dptr, &gpu_attr);
```

Comment:
I would do the pointer query inside `MPL_gpu_get_buffer_bounds`. It might be cuda version specific, so it makes sense to do it inside MPL wrappers.

Reply:
If we do the query in `MPL_gpu_get_buffer_bounds`, the caller might end up doing a redundant query. That is unless we modify the API to also return the device handle or pointer attributes. But if we return it through the API and the backend didn't require setting the device, we could end up doing the query for no reason.

Right now the 2 call sites in MPICH are already doing a pointer attribute query, so whichever design we choose is kind of moot.